state/protocol/snapshot.go

```diff
@@ -71,6 +71,13 @@ type Snapshot interface {
 	// are NOT guaranteed to have been validated by HotStuff.
 	Pending() ([]flow.Identifier, error)
 
+	// ValidPending returns the IDs of all descendants of the Head block.
+	// Valid means the returned bocks have at least one child.
+	// The IDs are ordered such that parents are included before their children. Includes
+	// only blocks that have child blocks.
+	// These are NOT guaranteed to have been validated by HotStuff.
+	ValidPending() ([]flow.Identifier, error)
```

Comment:
In my mind, pending has the meaning of "not finalized". But this method can also include finalized blocks, depending on the snapshot. Can we call this method `ValidDescendants` ? 

Would be great to also rename https://github.com/onflow/flow-go/blob/63052932c6c522989ab722ef931d2862001286ca/state/protocol/snapshot.go#L72 to `Descendants`